### PR TITLE
Add & apply pre-commit and lint job

### PR DESCRIPTION
**Pull request by jwodder**

### CODE REVIEW

**Comment by jwodder** [comment]
@nipy/team-heudiconv Please review this PR and comment any desired configuration changes.  I am using this as the base for several other PRs, so it would be good to get this PR processed quickly.

File: tox.ini [hunk 1]
```diff
@@ -1,6 +1,34 @@
 [tox]
-envlist = py27,py33,py34,py35
+envlist = lint,py27,py33,py34,py35
```
Comment (asmacdo) [inline tox.ini:2]: These versions don't line up with our requirements. 

IMO something to fix separately, lmk if you'd like me to file an issue.

File: tox.ini [hunk 2]
```diff
@@ -1,6 +1,34 @@
...
+commands =
+    flake8 heudiconv
+
+[flake8]
+doctests = True
+exclude = .*/,build/,dist/,test/data,venv/
+hang-closing = False
+unused-arguments-ignore-stub-functions = True
+select = A,B,B902,C,E,E242,F,U100,W
+ignore = A003,B005,E203,E262,E266,E501,W503
```
Comment (asmacdo) [inline tox.ini:24]: Nitpick (shouldn't block merge) 

IMO it's convenient to comment what these are.

**Review by asmacdo — APPROVED** [review]
Looks good to me!

File: tox.ini (hunk 1 above)
Comment (jwodder) [reply tox.ini:2]: That's handled by #660.

**Comment by asmacdo** [comment]
IMHO, I prefer a linter job to a precommit hook. (I haven't used them extensively, so please correct me if I'm missing something.)

The reason is that I sometimes want to commit and push a WIP (for example, to commit where I am so I can switch branches to reproduce a big) and I wouldn't want the linter failure to slow me down.

**Comment by jwodder** [comment]
@asmacdo 

> The reason is that I sometimes want to commit and push a WIP (for example, to commit where I am so I can switch branches to reproduce a bug) and I wouldn't want the linter failure to slow me down.

That's what `git commit`'s `-n`/`--no-verify` option is for.

**Comment by jwodder** [comment]
@yarikoptic It will cause conflicts, but that's only a problem for me.

**Pull request merged**